7a_temp_coop_munge/src/data_parsers/parse_missouri_files.R

```diff
@@ -0,0 +1,85 @@
+
+# Parse the ~300 files within the `Bull_Shoals_Lake_DO_and_Temp.zip` file
+
+parse_Bull_Shoals_Lake_DO_and_Temp <- function(inind, outind) {
+
+  infile <- sc_retrieve(inind, remake_file = '6_temp_coop_fetch_tasks.yml')
+  outfile <- as_data_file(outind)
+
+  # unzip the file with all of Bull Shoals
+  files_from_zip <- unzip(infile, exdir = tempdir())
+
+  # Read and clean each file (create on big file)
+  data_clean <- purrr::map(files_from_zip, function(fn) {
+
+    # Only read in the data part of the Excel file
+    # Assumes each file has only these rows/columns
+    # of data (which looked correct based on my spot check)
+    readxl::read_excel(fn, range = "A7:D27") %>%
+
+      # Struggles with the special symbols in the column names
+      # due to encoding issues. See details in the `scipiper`
+      # issue: https://github.com/USGS-R/scipiper/issues/151
+      fix_colname_encoding("latin1") %>%
+
+      # Reformat columns and extract info we need from Excel file
+      dplyr::mutate(DateTime = extract_bullshoals_date(fn),
+                    site = extract_bullshoals_site(fn),
+                    # TODO: Don't manually add the nhdhr id
+                    # Only doing now bc we don't have hypso for
+                    # Bull Shoals yet and I found this nhdhr match:
+                    # https://github.com/USGS-R/lake-temperature-model-prep/blob/master/4_params_munge.yml#L60
+                    id = 'nhdhr_12003253',
```

Comment:
this is probably convention in the `coop_parsers`, but we call this `site_id` not `id` elsewhere in the repo. But no need to change this if the convention is used in this section for some reason.

Reply:
I was following the convention listed [here](https://docs.google.com/document/d/1Z1fg8ng9IWHS9oiQTxsDvpGBlbgz2bMZ5rWNxfRiHC8/edit), where `id` is used for the column that can be crosswalked to an NHDID later.

Reply:
Gotcha, let's keep as-is for that convention. Thanks